block_engine/src/block_engine.rs

```diff
@@ -616,33 +629,42 @@ impl BlockEngineRelayerHandler {
         accounts_of_interest: &mut TimedCache<Pubkey, u8>,
         programs_of_interest: &mut TimedCache<Pubkey, u8>,
         address_lookup_table_cache: &DashMap<Pubkey, AddressLookupTableAccount>,
+        ofac_addresses: &HashSet<Pubkey>,
     ) -> Option<ExpiringPacketBatch> {
-        let filtered_packets: Vec<ProtoPacket> = block_engine_batches
-            .packet_batches
-            .into_iter()
-            .flat_map(|b| {
-                b.iter()
-                    .filter_map(|p| {
-                        let tx: VersionedTransaction = p.deserialize_slice(..).ok()?;
-
-                        let is_forwardable =
-                            is_aoi_in_static_keys(&tx, accounts_of_interest, programs_of_interest)
-                                || is_aoi_in_lookup_table(
-                                    &tx,
-                                    accounts_of_interest,
-                                    programs_of_interest,
-                                    address_lookup_table_cache,
-                                );
-
-                        if is_forwardable {
-                            Some(packet_to_proto_packet(p)?)
-                        } else {
-                            None
+        let mut filtered_packets = Vec::new();
+
+        for batch in &block_engine_batches.banking_packet_batch.0 {
+            for packet in batch {
+                if packet.meta().discard() {
+                    continue;
+                }
+
+                if !ofac_addresses.is_empty() {
+                    if let Ok(tx) = packet.deserialize_slice::<VersionedTransaction, _>(..) {
+                        if !is_tx_ofac_related(&tx, ofac_addresses, address_lookup_table_cache)
+                            && (is_aoi_in_static_keys(
+                                &tx,
+                                accounts_of_interest,
+                                programs_of_interest,
+                            ) || is_aoi_in_lookup_table(
+                                &tx,
+                                accounts_of_interest,
+                                programs_of_interest,
+                                address_lookup_table_cache,
+                            ))
+                        {
+                            if let Some(packet) = packet_to_proto_packet(packet) {
+                                filtered_packets.push(packet)
+                            }
                         }
-                    })
-                    .collect::<Vec<ProtoPacket>>()
-            })
-            .collect::<Vec<ProtoPacket>>();
+                    }
+                } else {
+                    if let Some(packet) = packet_to_proto_packet(packet) {
```

Comment:
shouldn't we still be checking is aoi here?  

seems like this way, if ofac_addresses empty, we forward everything to block engine.

Reply:
oh yeah good call

Reply:
fixed